homeassistant/components/group/entity.py

```diff
@@ -437,15 +476,28 @@ def _async_update_group_state(self, tr_state: State | None = None) -> None:
             self._state = None
             return
         # If the entity domains have more than one
-        # on state, we use STATE_ON/STATE_OFF
+        # on state, we use STATE_ON/STATE_OFF, unless there is
+        # only one specific `on` state in use for one specific domain
+        elif self.single_active_domain and len(active_on_states) == 1:
```

Comment:
If we know it's a single domain via `self.single_active_domain` don't we have the correct on states in `self._on_states` instead of making a new `active_on_states`?

Reply:
self._on_states is based on the off states of the entities found. `active_on_states` is based on the actual states.

Reply:
But you may be right. If you have suggestion for some rework, feel free to commit